### PR TITLE
Optimize the Split page

### PR DESCRIPTION
**Pull request by yhwang**
## Description
Reduce a `setTimeout()` call and optimize the loading widget rendering. Make the loading widget behavior the same as other pages. The timeline keeps updating data while in the loading phase.

related to: #21062 item 1

## Motivation and Context
When making the `Split` page have the same layout as other pages(Overview, Live Plan, and Stage Performance), one of
the comments is to optimize the loading widget and reduce a `setTimeout()` call. This PR addresses the comment and also makes the timeline visible while the query is still running, similar to the `Stage Performance` page.

## Impact
Users can see the timeline of splits while the query is still running.

## Test Plan
Verified the `Split` page when a query is running and is done.

## Contributor checklist

- [x] Please make sure your submission complies with our [development](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#development), [formatting](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#formatting), [commit message](https://github.com/prestodb/presto/wiki/Review-and-Commit-guidelines#commit-formatting-and-pull-requests), and [attribution guidelines](https://github.com/prestodb/presto/wiki/Review-and-Commit-guidelines#attribution).
- [x] PR description addresses the issue accurately and concisely.  If the change is non-trivial, a GitHub Issue is referenced.
- [ ] Documented new properties (with its default value), SQL syntax, functions, or other functionality.
- [ ] If release notes are required, they follow the [release notes guidelines](https://github.com/prestodb/presto/wiki/Release-Notes-Guidelines).
- [ ] Adequate tests were added if applicable.
- [ ] CI passed.

## Release Notes
Please follow [release notes guidelines](https://github.com/prestodb/presto/wiki/Release-Notes-Guidelines) and fill in the release notes below.

```
== NO RELEASE NOTE ==
```

### CODE REVIEW

**Comment by yhwang** [comment]
Enclose some screenshots of the Split page after the change:

while a query is running:
Screenshot 2024-01-11 at 3 24 50 PM
The loading widget is spinning and the timeline renders partial results.


When a query is done:
Screenshot 2024-01-11 at 3 26 50 PM
The loading widget is gone and the timeline renders the complete results.

File: presto-main/src/main/resources/webapp/src/components/Splits.jsx [hunk 1]
```diff
@@ -124,41 +150,18 @@ export default function Split(): void {
...
                     <h4><div className="loader">Loading...</div></h4>
                     )}
                     </div>
-                </div>
-            </>
-        );
-    }
-    // need to update the <div ref={containerRef} /> with the
-    // Timeline after the div is created
-    setTimeout(() => {
```
Comment (yhwang) [inline presto-main/src/main/resources/webapp/src/components/Splits.jsx:147]: this is the `setTimeout()` that could be optimized out.

**Comment by yhwang** [comment]
Hi @alonsovb, can your team help to review this change? Thanks!

File: presto-main/src/main/resources/webapp/src/components/Splits.jsx [hunk 2]
```diff
@@ -58,47 +58,73 @@ export default function Split(): void {
...
             clearTimeout(timerid.current);
             timerid.current = 0;
         }
-        const newQueryState = {query, ended: query.finalQueryInfo, failed: false, groups, items};
+        const newQueryState = {query, ended: query.finalQueryInfo, failed: false};
         if (newQueryState.ended === false && newQueryState.failed === false && timerid.current === 0) {
             timerid.current = setTimeout(queryResult, 3000);
         }
+        // use new data to update timeline
+        if (containerRef.current) {
```
[the remark this answers is not in the document]
Comment (yhwang) [reply presto-main/src/main/resources/webapp/src/components/Splits.jsx]: I put this util function inside `calculateItemsGroups` where the function is used only, therefore, no parameter passing.

[the remark this answers is not in the document]
Comment (aarong506) [reply presto-main/src/main/resources/webapp/src/components/Splits.jsx]: Great 👏🏽

**Review by jorgeramirezamora — APPROVED** [review]
LGTM!

**Review by tdcmeehan — APPROVED** [review]

**Pull request merged**